ip_test.go

```diff
@@ -1,235 +1,605 @@
 package echo
 
 import (
+	"github.com/stretchr/testify/assert"
 	"net"
 	"net/http"
-	"strings"
 	"testing"
-
-	testify "github.com/stretchr/testify/assert"
 )
 
-const (
-	// For RemoteAddr
-	ipForRemoteAddrLoopback  = "127.0.0.1" // From 127.0.0.0/8
-	sampleRemoteAddrLoopback = ipForRemoteAddrLoopback + ":8080"
-	ipForRemoteAddrExternal  = "203.0.113.1"
-	sampleRemoteAddrExternal = ipForRemoteAddrExternal + ":8080"
-	// For x-real-ip
-	ipForRealIP = "203.0.113.10"
-	// For XFF
-	ipForXFF1LinkLocal = "169.254.0.101" // From 169.254.0.0/16
-	ipForXFF2Private   = "192.168.0.102" // From 192.168.0.0/16
-	ipForXFF3External  = "2001:db8::103"
-	ipForXFF4Private   = "fc00::104" // From fc00::/7
-	ipForXFF5External  = "198.51.100.105"
-	ipForXFF6External  = "192.0.2.106"
-	ipForXFFBroken     = "this.is.broken.lol"
-	// keys for test cases
-	ipTestReqKeyNoHeader             = "no header"
-	ipTestReqKeyRealIPExternal       = "x-real-ip; remote addr external"
-	ipTestReqKeyRealIPInternal       = "x-real-ip; remote addr internal"
-	ipTestReqKeyRealIPAndXFFExternal = "x-real-ip and xff; remote addr external"
-	ipTestReqKeyRealIPAndXFFInternal = "x-real-ip and xff; remote addr internal"
-	ipTestReqKeyXFFExternal          = "xff; remote addr external"
-	ipTestReqKeyXFFInternal          = "xff; remote addr internal"
-	ipTestReqKeyBrokenXFF            = "broken xff"
-)
+func mustParseCIDR(s string) *net.IPNet {
+	_, IPNet, err := net.ParseCIDR(s)
+	if err != nil {
+		panic(err)
+	}
+	return IPNet
+}
+
+func TestIPChecker_TrustOption(t *testing.T) {
+	var testCases = []struct {
+		name         string
+		givenOptions []TrustOption
+		whenIP       string
+		expect       bool
+	}{
+		{
+			name: "ip is within trust range, trusts additional private IPV6 network",
+			givenOptions: []TrustOption{
+				TrustLoopback(false),
+				TrustLinkLocal(false),
+				TrustPrivateNet(false),
+				// this is private IPv6 ip
+				// CIDR Notation: 	2001:0db8:0000:0000:0000:0000:0000:0000/48
```

Comment:
For better readability for different tab sizes better uses spaces for aligning the values here (github also displays with tabsize of 4).
Not really important though.

Reply:
seems that Echo `.editorconfig` sets `indent_size = 2`